job-runner/client.go

```diff
@@ -41,52 +43,64 @@ func NewClient(c *http.Client, baseUrl string) Client {
 	}
 }
 
-func (c *client) NewJobChain(jobChain proto.JobChain) error {
+func (c *client) NewJobChain(jobChain proto.JobChain) (string, error) {
```

Comment:
This is a backwards-incompatible change. I.e. it'll break all code using `spincycle.Client`, which for us is a fair amount of code. We might have to +1 the minor version to control a roll out of this while we upgrade our code. Unless we can avoid making this breaking change?